tasks/main.yml

```diff
@@ -16,26 +16,33 @@
 
 - name: apply kubectl service to create inception service
   command: |
-    {{bin_dir}}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
+    {{kubectl_bin_dir}}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
   with_items: "{{ inception_files }}"
 
-- name: copy over jupyter notebook server config files
+- name: copy over jupyter notebook deployment config
   copy:    
-    src: "{{ item }}"
+    src: k8s-tf-nb-deployment.yml 
     dest: "{{ kube_config_dir }}/"
-  with_items: "{{ tf_nb_files }}"
 
+- name: copy over jupyter notebook service config
+  template:    
+    src: k8s-tf-nb-service.yml.j2
+    dest: "{{ kube_config_dir }}/k8s-tf-nb-service.yml"
 
 - name: apply kubectl service to create notebook service
   command: |
-    {{ bin_dir }}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
+    {{ kubectl_bin_dir }}/kubectl apply -f {{ kube_config_dir }}/{{ item }}
   with_items: "{{ tf_nb_files }}"
 
 - name: wait for inception service to start on port 30000
-  wait_for: port=30000
+  wait_for:
+    host: "{{ hostvars[groups['kube-node'][0]]['ansible_host'] }}"
+    port: 30000
 
 - name: wait for notebook server to start on port 31000
-  wait_for: port=31000
+  wait_for:
+    host: "{{ hostvars[groups['kube-node'][0]]['ansible_host'] }}"
+    port: 31000
 
 - name: copy admin-user file
```

Comment:
Is there a reason we're using an admin user?

Reply:
I was getting errors as a user I created myself but I probably did not set the right privileges. I left it as admin as this was just a demo.

Reply:
Looks like I could have done with reading this page: https://kubernetes.io/docs/admin/authorization/rbac/